sql/src/main/java/org/opensearch/sql/sql/SQLService.java

```diff
@@ -75,6 +91,83 @@ private AbstractPlan plan(
                     .isExplain(request.isExplainRequest())
                     .build()));
 
+    // There is no full support for JSON format yet for in memory operations, aliases, literals, and casts
+    // Aggregation has differences with legacy results
+    if (request.format().getFormatName().equals("json")) {
+      class VisitorContext {
+        @Getter
+        @Setter
+        boolean isJSONSupported = true;
+      }
+
+      class JsonSupportedVisitor extends AbstractNodeVisitor<Boolean, VisitorContext> {
+        @Override
+        public Boolean visit(Node node, VisitorContext context) {
+          // A node is supported if all of its children are supported.
+          return node.getChild().stream().filter(c -> c.accept(this, context) != null)
+              .allMatch(c -> c.accept(this, context));
```

Comment:
Twice `accept` for each sub-node? I think you can simplify this:
```
def visit(node):
    return visitChildren(node)
```
parent class has `visitChildren` which does all you need (recursive visiting & accumulating result). Maybe you need to override `aggregateResult` function.

Reply:
Thanks, fixed in f6db3453e5b8aabdf10a6ba12016e311d0835c6b